src/test/integration/IntegrationBase.t.sol

```diff
@@ -871,6 +887,23 @@ abstract contract IntegrationBase is IntegrationDeployer {
                                 UTILITY METHODS
     *******************************************************************************/
 
+    function _randomMagnitudes(uint64 sum, uint256 len) internal returns (uint64[] memory magnitudes) {
```

Comment:
Would recommend renaming this to something like `_randomMagnitudeAllocations` or `_randomAllocations` and also adding a `_randomMaxMagnitudes`.
This function can randomly assign initial maxMagnitude values. It could also optionally update maxMagnitude values to new maxMagnitudes per Strategy which are less than before (as a result of slashing).

Slashing is another event that can take place in the integration test flow which can occur prior, during, and after a staker's deposit and withdraw process and we'll want to play around with such testing scenarios.

Reply:
How would `_randomMaxMagnitudes` differ?